src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_MCAST_H
+#define UCC_MCAST_H
+
+#include <infiniband/ib.h>
+#include <infiniband/umad.h>
+#include <infiniband/verbs.h>
+#include <rdma/rdma_verbs.h>
+#include "utils/ucc_list.h"
+#include "utils/ucc_mpool.h"
+#include "components/tl/ucc_tl.h"
+#include "components/tl/ucc_tl_log.h"
+#include "utils/ucc_rcache.h"
+
+
+#define UCC_TL_MLX5_MCAST_ENABLE_BLOCKING true
+
+typedef struct mcast_coll_comm_init_spec {
+} mcast_coll_comm_init_spec_t;
+
+typedef struct ucc_tl_mlx5_mcast_lib {
+} ucc_tl_mlx5_mcast_lib_t;
+UCC_CLASS_DECLARE(ucc_tl_mlx5_mcast_lib_t, const ucc_base_lib_params_t *,
+                  const ucc_base_config_t *);
+
+typedef struct ucc_tl_mlx5_mcast_ctx_params {
+} ucc_tl_mlx5_mcast_ctx_params_t;
+
+typedef struct mcast_coll_context_t {
+} mcast_coll_context_t;
+
+typedef struct ucc_tl_mlx5_mcast_context_t {
+} ucc_tl_mlx5_mcast_context_t;
+
+
+typedef struct mcast_coll_comm { /* Stuff at a per-communicator sort of level */
+} mcast_coll_comm_t;
+
+typedef struct ucc_tl_mlx5_mcast_team {
+    void *mcast_comm;
```

Comment:
can you make it a pointer to `mcast_coll_comm_t` ?

Reply:
@samnordmann It is going away with new commits. Currently, it is an empty struct.